File: pinyougou-web/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/GoodsController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.common.pojo.PageResult;
import com.pinyougou.pojo.Goods;
import com.pinyougou.service.GoodsService;
import org.apache.commons.lang3.StringUtils;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * GoodsController
 *
 * @version 1.0
 * @date 2019/4/6
 */
@RestController
@RequestMapping("/goods")
public class GoodsController {

    @Reference(timeout = 10000)
    private GoodsService goodsService;

    @GetMapping("/findByPage")
    public PageResult findByPage(Goods goods, Integer page, Integer rows) {
        try {
            goods.setAuditStatus("0");
            if (StringUtils.isNotBlank(goods.getGoodsName())) {
                goods.setGoodsName(new String(goods.getGoodsName().getBytes("ISO8859-1"),"UTF-8"));
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return goodsService.findByPage(goods, page, rows);
    }

    @GetMapping("/updateStatus")
    public Boolean updateStatus(Long[] ids, String status) {
        try {
            goodsService.updateStatus(ids, status);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }
}

File: pinyougou-web/pinyougou-manager-web/src/main/webapp/js/controller/goodsController.js
app.controller("goodsController", function ($scope, baseService, $controller) {
    $controller("baseController", {$scope: $scope});

    $scope.status = ["未审核","已通过","审核不通过","关闭"];

    $scope.findByPage = function () {
        baseService.findByPage("/goods/findByPage?page=" + $scope.paginationConf.currentPage +
            "&rows=" + $scope.paginationConf.itemsPerPage, $scope.data).then(function (value) {
                $scope.dataList = value.data.rows;
                $scope.paginationConf.totalItems = value.data.total;
        })
    };

    $scope.updateStatus = function (status) {
        if ($scope.ids.length != 0) {
            baseService.doGet("/goods/updateStatus?ids="+ $scope.ids +"&status=" + status).then(function (value) {
                if (value.data) {
                    $scope.findByPage();
                    $scope.ids = [];
                } else {
                    alert("审核失败!");
                }
            })
        } else {
            alert("请选择要删除的数据!");
        }
    };

});